### PR TITLE
new NLP and ML Quickstarts with Snowpark and Streamlit submission

### PR DESCRIPTION
**Pull request by sfc-gh-imehaddi**

### CODE REVIEW

**Comment by sfc-gh-ghernandez** [comment]
@iamontheinet Can you take a look at this Quickstart by @sfc-gh-imehaddi please? This is a candidate for the demo program too (in time for Summit). Thank you.

File: site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md [hunk 1]
```diff
@@ -0,0 +1,395 @@
+author: sfc-gh-imehaddi
+id: end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis
+summary: This is a sample Snowflake Guide
+categories: Getting-Started
+environments: web
+status: Published 
+feedback link: https://github.com/Snowflake-Labs/sfguides/issues
+tags: Getting Started, Data Science, Data Engineering, Twitter 
+
+# End to End NLP and ML using Snowpark Python and Streamlit: Sentiments Analysis
```
Comment (sfc-gh-ghernandez) [inline site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md]:
```suggestion
NLP and ML with Snowpark Python and Streamlit for Sentiment Analysis
```

File: site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+<!-- ------------------------ -->
+## Overview 
+Duration: 5
+
+This quickstarts is showing how we can do Natural Language Processing (NLP) and ML within 100% Snowflake using Snowpark Python ans Streamlit as well. Our use case is about doing : Sentiment Analysis with 100% Snowpark (feature engineering, train and prediction).
```
Comment (sfc-gh-ghernandez) [inline site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md]:
```suggestion
This Quickstart will demonstrate how you can perform Natural Language Processing (NLP) and ML within Snowflake using Snowpark Python and Streamlit. We'll use these tools to perform sentiment analysis with Snowpark (feature engineering, training, and prediction).
```

File: site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+- Do NLP and ML on Snowflake using Snowpark
+- Load data into Snowflake
+- Transform your data using Snowpark DataFrame API
+- Train a scikit-learn model using Store Procedure inside Snowflake
+- Deploy a model using UDF Function
+- Inference with UDF Function
+- Use Streamlit with Snowpark
+
+### What You’ll Need 
+- A Snowflake Account with ACCOUNTADMIN role. If you do'nt have one, you can register for a [free trial account](https://signup.snowflake.com/) Account
```
Comment (sfc-gh-ghernandez) [inline site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md]:
```suggestion
- A Snowflake Account with ACCOUNTADMIN role. If you don't have one, you can register for a [free trial account](https://signup.snowflake.com/)
```

File: site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+### Clone GitHub repository
+First, clone the source code for this repo to your local environment:
+
+```shell
+git clone https://github.com/Snowflake-Labs/snowpark-python-demos.git
+cd snowpark_nlp_ml_demo/
+```
+
+### Setup Python Environment:
+Create conda environment. Let's name the environment nlp_ml_sentiment_analysis.
```
Comment (sfc-gh-ghernandez) [inline site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md]:
```suggestion
Create a conda environment. Let's name the environment nlp_ml_sentiment_analysis.
```

File: site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md [hunk 5]
```diff
@@ -0,0 +1,395 @@
...
+
+```shell
+conda update conda
+conda update python
+conda env create -f ./snowpark-env/conda-env_nlp_ml_sentiment_analysis.yml  --force
+```
+
+### Snowflake Credentials
+Update the Snowflake connexion file : connection.json
```
Comment (sfc-gh-ghernandez) [inline site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md]:
```suggestion
Update the Snowflake connection file: connection.json
```

File: site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md [hunk 6]
```diff
@@ -0,0 +1,395 @@
...
+    "user": "",
+    "password": "",
+    "role": "ACCOUNTADMIN",
+    "database": "IMDB",
+    "schema": "PUBLIC",
+    "warehouse": "DEMO_WH"
+ }
+```
+
+### Activate Python environment using conda :
```
Comment (sfc-gh-ghernandez) [inline site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md:80]:
```suggestion
### Activate Python environment using conda
```

File: site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md [hunk 7]
```diff
@@ -0,0 +1,395 @@
...
+```shell
+conda activate nlp_ml_sentiment_analysis
+```
+
+### Run Streamlit App :
```
Comment (sfc-gh-ghernandez) [inline site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md:85]:
```suggestion
### Run Streamlit App
```

File: site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md [hunk 8]
```diff
@@ -0,0 +1,395 @@
...
+```shell
+cd notebook
+jupyter notebook
+```
+
+<!-- ------------------------ -->
+## Snowflake Environment Setup
+Duration: 2
+
+### Option 1 - Automatically : with Streamlit App
```
Comment (sfc-gh-ghernandez) [inline site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md:102]:
```suggestion
### Option 1 - Environment setup via the Streamlit App
```

File: site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md [hunk 9]
```diff
@@ -0,0 +1,395 @@
...
+Use the Streamlit App to setup Snowflake Objects
+
+![Puppy](./assets/Setup_Snowflake_Objects.png)
```
Comment (sfc-gh-ghernandez) [inline site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md:105]: Update all instances of "Puppy" in here to a proper description of the image. This is for visually impaired users who use screen readers to navigate the internet.

File: site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md [hunk 10]
```diff
@@ -0,0 +1,395 @@
...
+
+Make sure you have this result :
+
+![Puppy](./assets/Setup_Snowflake_Objects_Results.png)
```
Comment (sfc-gh-ghernandez) [inline site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md:109]: Update all instances of "Puppy" in here to a proper description of the image. This is for visually impaired users who use screen readers to navigate the internet.

File: site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md [hunk 11]
```diff
@@ -0,0 +1,395 @@
...
+
+You can check directly with Snowsight that the data are available in Snowflake.
+
+![Puppy](./assets/DB_IMDB.png)
```
Comment (sfc-gh-ghernandez) [inline site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md:113]: Update all instances of "Puppy" in here to a proper description of the image. This is for visually impaired users who use screen readers to navigate the internet.

File: site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md [hunk 12]
```diff
@@ -0,0 +1,395 @@
...
+
+### Option 2 - Manually : with Snowsight
```
Comment (sfc-gh-ghernandez) [inline site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md:115]:
```suggestion
### Option 2 - Environment setup via Snowsight
```

File: site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md [hunk 13]
```diff
@@ -0,0 +1,395 @@
...
+First, log into your Snowflake Account (Snowsight Web UI) using your credentials.
+
+Then, run the following SQL commands to create the DATABASE :
```
Comment (sfc-gh-ghernandez) [inline site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md:118]:
```suggestion
Then, run the following SQL commands to create the database:
```

File: site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md [hunk 14]
```diff
@@ -0,0 +1,395 @@
...
+
+```sql
+USE ROLE ACCOUNTADMIN;
+
+CREATE DATABASE if not EXISTS IMDB;
+```
+
+Run the following SQL commands to create the TABLES :
```
Comment (sfc-gh-ghernandez) [inline site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md:126]:
```suggestion
Run the following SQL commands to create the tables:
```

File: site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md [hunk 15]
```diff
@@ -0,0 +1,395 @@
...
+	SENTIMENT STRING
+);
+
+CREATE TABLE if not EXISTS TEST_DATASET (
+	REVIEW STRING,
+	SENTIMENT STRING
+);
+```
+
+Run the following SQL commands to create the WAREHOUSE :
```
Comment (sfc-gh-ghernandez) [inline site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md:142]:
```suggestion
Run the following SQL commands to create the warehouse:
```

File: site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md [hunk 16]
```diff
@@ -0,0 +1,395 @@
...
+```sql
+CREATE WAREHOUSE if not EXISTS DEMO_WH WAREHOUSE_SIZE=MEDIUM INITIALLY_SUSPENDED=TRUE AUTO_SUSPEND=120;
+```
+
+Run the following SQL commands to create the STAGE :
```
Comment (sfc-gh-ghernandez) [inline site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md:147]:
```suggestion
Run the following SQL commands to create the stage:
```

File: site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md [hunk 17]
```diff
@@ -0,0 +1,395 @@
...
+
+USE IMDB.PUBLIC;
+```
+
+
+<!-- ------------------------ -->
+## Load Data
+Duration: 5
+
+We used Python code to load the data into Snowflake. In order to simplify code execution you can click on the right button to start loading the data.
```
Comment (sfc-gh-ghernandez) [inline site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md:159]: Which button does "right button" refer to? I was trying to locate it via the screenshot, but did not know which one you were referring to.

File: site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md [hunk 18]
```diff
@@ -0,0 +1,395 @@
...
+    session.write_pandas(pandas_df, "TRAIN_DATASET", auto_create_table=False, overwrite=True)
+```
+
+<!-- ------------------------ -->
+## Analyze Data
+Duration: 2
+
+### What You'll Do :
+
+Use use the section **Analyze** to explore and analyze the datasets and see some metrics.
```
Comment (sfc-gh-ghernandez) [inline site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md]:
```suggestion
Use the section **Analyze** to explore and analyze the datasets and see some metrics.
```

File: site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md [hunk 19]
```diff
@@ -0,0 +1,395 @@
...
+
+![Puppy](./assets/Analyze_Data1.png)
```
Comment (sfc-gh-ghernandez) [inline site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md:197]: Same comment about "Puppy", and for all instances below.

File: site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md [hunk 20]
```diff
@@ -0,0 +1,395 @@
...
+
+#### Select your data
+Choose the dataset that you want to analyze :
```
Comment (sfc-gh-ghernandez) [inline site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md]: A space is not needed between the final word of the statement and `:`. Can you please update this in the entire file?

```suggestion
Choose the dataset that you want to analyze:
```

File: site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md [hunk 21]
```diff
@@ -0,0 +1,395 @@
...
+You can see a sample of data :
+
+![Puppy](./assets/Sample_data.png)
+
+#### Data Description
+Here a description of your dataset :
+
+![Puppy](./assets/Data_Description.png)
+
+### What You'll Learn :
```
Comment (sfc-gh-ghernandez) [inline site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis.md:219]: You don't need a `:` at the end of section titles (i.e. titles that start with `#`, `##`, etc,), so you can remove from here and rest of file.

File: site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/assets/SAMPLE.jpg
Comment (sfc-gh-ghernandez) [inline site/sfguides/src/end_to_end_nlp_and_ml_using_snowpark_python_and_streamlit:_sentiments_analysis/assets/SAMPLE.jpg:1]: You can delete this file, just to make the overall PR smaller.

**Review by sfc-gh-ghernandez — CHANGES_REQUESTED** [review]
Hi @sfc-gh-imehaddi – I requested some very minor changes, and also left some comments where you should update things in the entire .md file. @iamontheinet may have more comments, but on my end, that is all I had for now. Thank you again for submitting!!

**Comment by sfc-gh-imehaddi** [comment]
@iamontheinet Can you merge and deploy this please? Cc @sfc-gh-ghernandez

**Comment by sfc-gh-imehaddi** [comment]
@iamontheinet Can you please check and merge this please? Cc @sfc-gh-ghernandez

**Review by sfc-gh-imehaddi — COMMENTED** [review]
Update following feedbacks. @iamontheinet

**Comment by sfc-gh-imehaddi** [comment]
No title on Dash side even if it's run locally

**Pull request closed without merging**